pkg/common/security/acl_test.go

```diff
@@ -19,119 +19,206 @@
 package security
 
 import (
+	"errors"
+	"fmt"
 	"testing"
-
-	"gotest.tools/assert"
 )
 
-func TestACLCreate(t *testing.T) {
-	_, err := NewACL("")
-	if err != nil {
-		t.Errorf("parsing failed for string: ''")
-	}
-	_, err = NewACL(" ")
-	if err != nil {
-		t.Errorf("parsing failed for string: ' '")
-	}
-	_, err = NewACL("user1")
-	if err != nil {
-		t.Errorf("parsing failed for string: 'user1'")
-	}
-	_, err = NewACL("user1,user2")
-	if err != nil {
-		t.Errorf("parsing failed for string: 'user1,user2'")
-	}
-	_, err = NewACL("user1,user2 ")
-	if err != nil {
-		t.Errorf("parsing failed for string: 'user1,user2 '")
-	}
-	_, err = NewACL("user1,user2 group1")
-	if err != nil {
-		t.Errorf("parsing failed for string: 'user1,user2 group1'")
-	}
-	_, err = NewACL("user1,user2 group1,group2")
-	if err != nil {
-		t.Errorf("parsing failed for string: 'user1,user2 group1,group2'")
-	}
-	_, err = NewACL("user2 group1,group2")
-	if err != nil {
-		t.Errorf("parsing failed for string: 'user2 group1,group2'")
-	}
-	_, err = NewACL(" group1,group2")
-	if err != nil {
-		t.Errorf("parsing failed for string: ' group1,group2'")
-	}
-	_, err = NewACL("* group1,group2")
-	if err != nil {
-		t.Errorf("parsing failed for string: '* group1,group2'")
-	}
-	_, err = NewACL("user1,user2 *")
-	if err != nil {
-		t.Errorf("parsing failed for string: 'user1,user2 *'")
-	}
-	_, err = NewACL("*")
-	if err != nil {
-		t.Errorf("parsing failed for string: '*'")
-	}
-	_, err = NewACL("* ")
-	if err != nil {
-		t.Errorf("parsing failed for string: '* '")
-	}
-	_, err = NewACL(" *")
-	if err != nil {
-		t.Errorf("parsing failed for string: ' *'")
+func IsSameACL(got, expected ACL) error {
+	if got.allAllowed != expected.allAllowed {
+		return errors.New("allAllowed is not same")
+	}
+	if expected.users != nil && got.users != nil {
+		for username, expectedAllow := range expected.users {
+			if gotAllow, ok := got.users[username]; !ok {
+				return fmt.Errorf("username %s does not exist", username)
+			} else if expectedAllow != gotAllow {
+				return fmt.Errorf("username %s is not same", username)
+			}
+		}
+	} else if (expected.users != nil && got.users == nil) || (expected.users == nil && got.users != nil) {
+		return errors.New("users of an acl is not nil, expect one and got one should be same")
```

Comment:
You are not catching the case that `got` has more users than `expected`
Simplify and change check ordering:
1. check if length of both maps are equal: if not return
2. check equality of the members: let the range handle nil maps. 
3. Combine the user not found and the allow flag difference in one check:
```
gotAllow, ok := got.users[username]; !ok || gotAllow != expectedAllow
```

Reply:
Hi @wilfred-s 
Understood, thank you for comments.
I am going to update the commit later.